build.gradle

```diff
@@ -63,6 +63,17 @@ subprojects {
         targetCompatibility = JavaVersion.VERSION_17
     }
 
+    compileTestJava {
+        options.encoding = 'UTF-8'
+
+        // Makes spotlessApply task run on every compile/build.
+        dependsOn 'spotlessApply'
+
+        // Nails the Java-Version of every Subproject
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+    }
```

Comment:
This is a direct copy of `compileJava`. Can we get rid of the duplication somehow (helper method or similar)?

@illuminator3 any idea?

Reply:
`compileTestJava` and `compileJava` both want a closure. You can just make a variable for it and then assign both to it in a single expression:
```groovy
def javaCompileThings = {
        options.encoding = 'UTF-8'

        // Makes spotlessApply task run on every compile/build.
        dependsOn 'spotlessApply'

        // Nails the Java-Version of every Subproject
        sourceCompatibility = JavaVersion.VERSION_17
        targetCompatibility = JavaVersion.VERSION_17
}

compileTestJava = compileJava = javaCompileThings
```I'm not sure as of right now if the above works, in case it doesn't, try the following:
```gradle
compileTestJava(javaCompileThings)
compileJava(javaCompileThings)
```

Reply:
Cool stuff, ill try

Reply:
Former didnt work:
> Cannot assign 'Closure' to 'JavaCompile' 

But latter did, thanks